cni/internal/cniutil.go

```diff
@@ -61,3 +64,41 @@ func IfacesWithName(name string, ifaces ...*current.Interface) []*current.Interf
 
 	return foundIfaces
 }
+
+// VMTapPair takes a CNI result and returns the tap iface and vm iface corresponding
+// to the provided vmID. See the vmconf package docs for details on the expected
+// vm and tap iface configurations.
+func VMTapPair(
+	result *current.Result, vmID string,
+) (vmIface *current.Interface, tapIface *current.Interface, err error) {
+	vmIfaces, otherIfaces := FilterBySandbox(vmID, result.Interfaces...)
+	if len(vmIfaces) > 1 {
+		return nil, nil, errors.Errorf(
+			"expected to find at most 1 interface in sandbox %q, but instead found %d",
+			vmID, len(vmIfaces))
+	}
+	if len(vmIfaces) == 0 {
```

Comment:
nit: this can be an else if

Reply:
Sure, I don't have a preference either way.